### PR TITLE
Update SyncJobIndex

### PR DESCRIPTION
**Pull request by wangch079**
This PR adds a `create` method to class `SyncJobIndex`, to create a new sync job.

#### Pre-Review Checklist
- [x] Covered the changes with automated tests
- [x] Tested the changes locally by running `make test` and `make ftest NAME=mysql`
- [x] Added a label for each target release version (example: `v7.13.2`, `v7.14.0`, `v8.0.0`)
- [ ] Considered corresponding documentation changes
- [ ] Contributed any configuration settings changes to the configuration reference

### CODE REVIEW

File: connectors/tests/test_byoc.py
Comment (timgrein) [inline connectors/tests/test_byoc.py:973]:
```suggestion
    index_method, sync_now, trigger_method, set_env
```

I think we only need `set_env` to set `elasticsearch.password` in the os environment. `patch_logger` seems unnecessary.

File: connectors/tests/test_byoc.py
Comment (timgrein) [inline connectors/tests/test_byoc.py:1000]:
```suggestion
async def test_pending_jobs(get_all_docs, set_env):
```

File: connectors/tests/test_byoc.py
Comment (timgrein) [inline connectors/tests/test_byoc.py:1033]:
```suggestion
async def test_orphaned_jobs(get_all_docs, set_env):
```

File: connectors/tests/test_byoc.py
Comment (timgrein) [inline connectors/tests/test_byoc.py:1052]:
```suggestion
async def test_stuck_jobs(get_all_docs, set_env):
```

**Review by timgrein — APPROVED** [review]
LGTM, added some suggestions to remove `patch_logger`. Doesn't seem necessary.

File: connectors/tests/test_byoc.py
Comment (wangch079) [reply connectors/tests/test_byoc.py:973]: `patch_logger` is to silent logs.

compare this (with `patch_logger`):
```
chenhui.wang@SG-Chenhui connectors-python % bin/pytest -sv connectors/tests/test_byoc.py -k test_create_job
============================================================================================================================ test session starts =============================================================================================================================
platform darwin -- Python 3.10.7, pytest-7.1.3, pluggy-1.0.0 -- /Users/chenhui.wang/src/github.com/elastic/connectors-python/bin/python3.10
cachedir: .pytest_cache
Using --randomly-seed=2671744692
rootdir: /Users/chenhui.wang/src/github.com/elastic/connectors-python
plugins: fail-slow-0.3.0, asyncio-0.19.0, randomly-3.12.0, cov-3.0.0
asyncio: mode=strict
collected 69 items / 67 deselected / 2 selected                                                                                                                                                                                                                              

connectors/tests/test_byoc.py::test_create_job[True-JobTriggerMethod.ON_DEMAND] PASSED
connectors/tests/test_byoc.py::test_create_job[False-JobTriggerMethod.SCHEDULED] PASSED

====================================================================================================================== 2 passed, 67 deselected in 0.24s ======================================================================================================================
```

with this (without `patch_logger`):
```
chenhui.wang@SG-Chenhui connectors-python % bin/pytest -sv connectors/tests/test_byoc.py -k test_create_job
============================================================================================================================ test session starts =============================================================================================================================
platform darwin -- Python 3.10.7, pytest-7.1.3, pluggy-1.0.0 -- /Users/chenhui.wang/src/github.com/elastic/connectors-python/bin/python3.10
cachedir: .pytest_cache
Using --randomly-seed=147054654
rootdir: /Users/chenhui.wang/src/github.com/elastic/connectors-python
plugins: fail-slow-0.3.0, asyncio-0.19.0, randomly-3.12.0, cov-3.0.0
asyncio: mode=strict
collected 69 items / 67 deselected / 2 selected                                                                                                                                                                                                                              

connectors/tests/test_byoc.py::test_create_job[False-JobTriggerMethod.SCHEDULED] [FMWK][17:57:24][INFO] Loading config from /Users/chenhui.wang/src/github.com/elastic/connectors-python/connectors/tests/config.yml
PASSED
connectors/tests/test_byoc.py::test_create_job[True-JobTriggerMethod.ON_DEMAND] [FMWK][17:57:24][INFO] Loading config from /Users/chenhui.wang/src/github.com/elastic/connectors-python/connectors/tests/config.yml
PASSED

====================================================================================================================== 2 passed, 67 deselected in 0.23s ======================================================================================================================
```

**Comment by wangch079** [comment]
This PR includes some changes in PR #440 , which is being blocked.

Will only merge this after #440 is merged.

File: connectors/tests/test_byoc.py
Comment (timgrein) [reply connectors/tests/test_byoc.py:973]: Ok, I see, thank you for explaining! Maybe it's worth to have a separate fixture `silent_logger` / `deactivate_logging`, which is only there for deactivating logging. It's not obvious that `patch_logger` has this side effect. Could be done in a separate PR.

File: connectors/byoc.py
Comment (artem-shelkovnikov) [inline connectors/byoc.py:356]: What's your plan with https://github.com/elastic/connectors-python/pull/440?

First merge the #440 and then rebase here, or this PR will handle all?

File: connectors/tests/test_byoc.py
Comment (artem-shelkovnikov) [reply connectors/tests/test_byoc.py:973]: `patch_logger` is implemented not to silence logs, but to be able to assert on logs. I don't think silent logs are good - they are instead making it much much harder to debug a problem once the test fails.

Comment (artem-shelkovnikov) [reply connectors/tests/test_byoc.py:973]: I'd love to change it to False eventually if nobody is against it

File: connectors/byoc.py
Comment (wangch079) [reply connectors/byoc.py:356]: Yes, I will rebase. I don't intend to change anything to class `PipelineSettings` in this PR

**Review by artem-shelkovnikov — APPROVED** [review]

**Pull request merged**
